src/Hyperion/Benchmark.hs

```diff
@@ -55,7 +60,37 @@ instance Show Benchmark where
   showsPrec x (WithSampling _opt bk) = showsPrec x bk
 
 bench :: String -> Batch () -> Benchmark
-bench name batch = Bench (Text.pack name) batch
+bench name batch = Bench Text.pack (const []) id (const batch) name
+
+benchWith
+  :: (a -> Text)
+  -> (a -> [JSON.Pair])
+  -> (a -> b)
+  -> (b -> Batch ())
+  -> a -> Benchmark
+benchWith = Bench
+
+benchMany
```

Comment:
> However we can't expect the user to parse it and go fish out the input element's show output.

That's true, but that's not a concern for the benchmarks DSL. The DSL is there to allow the user to express her intent. We are then free to interpret that DSL in any way we want. If currently the input size in `series` gets munged in the benchmark name and that particular piece of metadata is therefore opaque to Kibana, then it sounds to me like that's not a DSL problem, that's an interpretation problem. Because after all, via `series` the DSL already lets the user express input sizes. So you do need to explain why fixing the interpretation won't work and why you therefore really do need to change the benchmarks DSL.